src/Hazelcast.Net.Linq.Async/Visitors/ColumnProjector.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Text;
+using Hazelcast.Linq.Evaluation;
+using Hazelcast.Linq.Expressions;
+
+namespace Hazelcast.Linq.Visitors
+{
+    /// <summary>
+    /// Traverse and creates column definitations for projection.
+    /// </summary>
+    internal class ColumnProjector : HzExpressionVisitor
+    {
+        public ExpressionNominator Nominator { get; }
+
+        private Dictionary<ColumnExpression, ColumnExpression> _mapOfColumns;
+        private List<ColumnDefinition> _columns;
+        private HashSet<string> _columnNames;
+        private HashSet<Expression> _candidates;
+        /// <summary>
+        /// Alias of the current level
+        /// </summary>
+        private string _existingAlias;
```

Comment:
maybe don't use `///` comments on private fields?

Reply:
Sure, why not. Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/719/commits/6dcef8a55d73060af97c494821575516c8cd240e